instrumentation-genai/opentelemetry-instrumentation-google-genai/src/opentelemetry/instrumentation/google_genai/generate_content.py

```diff
@@ -129,21 +131,60 @@ def _determine_genai_system(models_object: Union[Models, AsyncModels]):
     return _get_gemini_system_name()
 
 
-def _get_config_property(
-    config: Optional[GenerateContentConfigOrDict], path: str
-) -> Any:
+def _to_dict(value: object):
+    if isinstance(value, dict):
+        return value
+    if hasattr(value, "model_dump"):
+        return value.model_dump()
+    return json.loads(json.dumps(value))
+
+
+def _add_request_options_to_span(
+    span, config: Optional[GenerateContentConfigOrDict]
+):
     if config is None:
-        return None
-    path_segments = path.split(".")
-    current_context: Any = config
-    for path_segment in path_segments:
-        if current_context is None:
-            return None
-        if isinstance(current_context, dict):
-            current_context = current_context.get(path_segment)
-        else:
-            current_context = getattr(current_context, path_segment)
-    return current_context
+        return
+    span_context = span.get_span_context()
+    if not span_context.trace_flags.sampled:
+        # Avoid potentially costly traversal of config
+        # options if the span will be dropped, anyway.
+        return
+    # Automatically derive attributes from the contents of the
+    # config object. This ensures that all relevant parameters
+    # are captured in the telemetry data (except for those
+    # that are excluded via "exclude_keys").
+    attributes = flatten_dict(
+        _to_dict(config),
```

Comment:
I saw you're planning to make a semconv PR for some of these parameters. Are you planning to declare `gen_ai.gcp.request` as dynamic or listing out the actual expected request params?

Reply:
Thanks. I created an issue for defining the Semantic Conventions.

https://github.com/open-telemetry/semantic-conventions/issues/2053

I will get this convention defined and update here before we complete this PR.

I think dynamic is the better way to go, what do you think?

Reply:
It should still be possible to align them with this approach, albeit if we treat the Gen AI SDK as the canonical representation of the config (rather than SemConv as the canonical representation of the config).

I think this makes sense given the approach that Gen AI SDK is taking towards unifying the mapping of the different config options.

Reply:
The definition is proposed in PR https://github.com/open-telemetry/semantic-conventions/pull/2125 .

Marking as resolved for now.